### PR TITLE
[1/3] Jest refactor independent changes

### PR DESCRIPTION
**Pull request by lachrist**
A mixbag of independent changes. Some are general improvements. Others specifically targets jest support. Most of them are refactoring. The only two minor behavior changes are:
- [feat: make status independent from uncaught error](https://github.com/getappmap/appmap-agent-js/commit/9c9a93f866a872853349ad98ef30a87bfe54eb32)
- [feat: more precise track termination](https://github.com/getappmap/appmap-agent-js/commit/ff1d54bf81415af65ccb6c9092ec1adfb05f5201)

### CODE REVIEW

File: components/hook-exit/node/index.test.mjs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import process from "node:process";
+import { assertDeepEqual } from "../../__fixture__.mjs";
+import { testHookAsync } from "../../hook-fixture/index.mjs?env=test";
+import * as HookError from "./index.mjs?env=test";
```
Comment (dividedmind) [inline components/hook-exit/node/index.test.mjs:4]: This should probably be called `HookExit`.

File: components/hook-fixture/fixture/index.mjs [hunk 2]
```diff
@@ -37,7 +37,11 @@ export const testHookAsync = async (
   try {
     recordStartTrack(agent, "record", {}, null);
     await callbackAsync();
-    recordStopTrack(agent, "record", { type: "manual" });
+    // TODO this completely breaks encapsulation
+    //      but hook-fixture is only used for testing...
```
Comment (dividedmind) [inline components/hook-fixture/fixture/index.mjs:41]: I think breaking encapsulation for tests is acceptable sometimes.

**Review by dividedmind — APPROVED** [review]
I had a couple of suggestions but looks ok in general :)

File: components/hook-fixture/fixture/index.mjs (hunk 2 above)
Comment (lachrist) [reply components/hook-fixture/fixture/index.mjs:41]: done better than perfect?

File: components/hook-exit/node/index.test.mjs (hunk 1 above)
Comment (lachrist) [reply components/hook-exit/node/index.test.mjs:4]: done in 990a60a4bf6e041822cb43e669b1f635b089d15b

File: components/backend/default/index.mjs [hunk 3]
```diff
@@ -47,12 +47,21 @@ export const sendBackend = (
...
+      }
+      tracks.clear();
+    } else {
+      assert(tracks.has(key), "missing track", InternalAppmapError);
+      assert(!traces.has(key), "duplicate trace", InternalAppmapError);
+      const messages = tracks.get(key);
+      messages.push(message);
+      tracks.delete(key);
+      traces.set(key, compileTrace(configuration, messages));
+    }
```
[the remark this answers is not in the document]
Comment (lachrist) [reply components/backend/default/index.mjs:63]: fixup 783a0c003a7416536cf263054ad4098ee78274ba

**Comment by lachrist** [comment]
@dividedmind thanks for the review! I will merge this once we are done with the next two branches.

File: components/hook-fixture/fixture/index.mjs (hunk 2 above)
Comment (dividedmind) [reply components/hook-fixture/fixture/index.mjs:41]: Yup :)

**Pull request merged**